gateway-ha/src/main/java/io/trino/gateway/ha/handler/QueryIdCachingProxyHandler.java

```diff
@@ -122,6 +135,26 @@ else if (path.startsWith(TRINO_UI_PATH)) {
         return queryId;
     }
 
+    String getBackendForRequest(HttpServletRequest request)
+    {
+        String routingGroup = routingGroupSelector.findRoutingGroup(request);
+        String user = getQueryUser(request);
+        if (!Strings.isNullOrEmpty(routingGroup)) {
+            // This falls back on adhoc backend if there are no cluster found for the routing group.
+            return routingManager.provideBackendForRoutingGroup(routingGroup, user);
+        }
+        else {
+            return routingManager.provideAdhocBackend(user);
+        }
+    }
+
+    private boolean doRecordQueryId(HttpServletRequest request)
```

Comment:
name is not clear what this method should do

recordQueryId

trackQueryId 

or so? 

having "do" and "record" is weird

Reply:
maybe think of what you would write for javadoc